crates/indexer/src/models/token_bcs_utils.rs

```diff
@@ -0,0 +1,123 @@
+use crate::models::property_map::PropertyMap;
+use aptos_api_types::Address;
+use bcs;
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::Value;
+
+/// Convert the bcs serialized vector<u8> to its original string format
+pub fn convert_bcs_hex(typ: String, value: String) -> Option<String> {
+    let decoded = hex::decode(value.strip_prefix("0x").unwrap_or(&*value)).ok()?;
+
+    match typ.as_str() {
+        "0x1::string::String" => bcs::from_bytes::<String>(decoded.as_slice()),
+        "u8" => bcs::from_bytes::<u8>(decoded.as_slice()).map(|e| format!("{}", e)),
+        "u64" => bcs::from_bytes::<u64>(decoded.as_slice()).map(|e| format!("{}", e)),
+        "u128" => bcs::from_bytes::<u128>(decoded.as_slice()).map(|e| format!("{}", e)),
+        "bool" => bcs::from_bytes::<bool>(decoded.as_slice()).map(|e| format!("{}", e)),
+        "address" => bcs::from_bytes::<Address>(decoded.as_slice()).map(|e| format!("{}", e)),
+        _ => Ok(value),
+    }
+    .ok()
+}
+
+/// Convert the vector<u8> that is directly generated from b"xxx"
+pub fn convert_hex(val: String) -> Option<String> {
+    let decoded = hex::decode(val.strip_prefix("0x").unwrap_or(&*val)).ok()?;
+    String::from_utf8(decoded).ok()
+}
+
+/// Convert the json serialized PropertyMap's inner BCS fields to their original value in string format
+pub fn convert_bcs_propertymap(s: Value) -> Option<String> {
+    match PropertyMap::from_bsc_encode_str(s) {
+        Some(e) => match serde_json::to_string(&e) {
+            Ok(val) => Some(val),
+            Err(_) => None,
+        },
+        None => None,
+    }
+}
+
+pub fn deserialize_string_from_hexstring<'de, D>(
+    deserializer: D,
+) -> core::result::Result<String, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = <String>::deserialize(deserializer)?;
+    Ok(convert_hex(s.clone()).unwrap_or(s))
+}
+
+pub fn deserialize_string_from_bcs_hexstring<'de, D>(
+    deserializer: D,
+) -> core::result::Result<String, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = <String>::deserialize(deserializer)?;
+    Ok(convert_bcs_hex("0x1::string::String".to_string(), s.clone()).unwrap_or(s))
+}
+
+
+/// convert the bcs encoded inner value of property_map to its original value in string format
+pub fn deserialize_property_map_from_bcs_hexstring<'de, D>(
+    deserializer: D,
+) -> core::result::Result<String, D::Error>
```

Comment:
Let's have this return Value.

Reply:
```
    let str = convert_bcs_propertymap(s.clone()).unwrap_or_else(|| s.to_string());
    serde_json::from_str(&str).map_err(serde::de::Error::custom)
```

Reply:
updated to return value instead of string